modules/light-clients/08-wasm/keeper/msg_server_test.go

```diff
@@ -88,11 +97,146 @@ func (suite *KeeperTestSuite) TestMsgStoreCode() {
+					store.Set(host.ClientStateKey(), clienttypes.MustMarshalClientState(suite.chainA.App.AppCodec(), expClientState))
+
+					data, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					return &wasmvmtypes.Response{Data: data}, wasmtesting.DefaultGasUsed, nil
+				}
+			},
+			nil,
+		},
+		{
+			"failure: same code hash",
+			func() {
+				newCodeHash = oldCodeHash[:]
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, newCodeHash, []byte("{}"))
+
+				suite.mockVM.MigrateFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					panic("unreachable")
+				}
+			},
+			types.ErrWasmCodeExists,
+		},
+		{
+			"failure: unauthorized signer",
+			func() {
+				msg = types.NewMsgMigrateContract(suite.chainA.SenderAccount.GetAddress().String(), defaultWasmClientID, newCodeHash, []byte("{}"))
+			},
+			ibcerrors.ErrUnauthorized,
+		},
+		{
+			"failure: invalid wasm code hash",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, []byte(ibctesting.InvalidID), []byte("{}"))
+			},
+			types.ErrWasmCodeHashNotFound,
+		},
+		{
+			"failure: invalid client id",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, ibctesting.InvalidID, newCodeHash, []byte("{}"))
+			},
+			clienttypes.ErrClientTypeNotFound,
+		},
+		{
+			"failure: contract returns error",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, newCodeHash, []byte("{}"))
+
+				suite.mockVM.MigrateFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					return nil, wasmtesting.DefaultGasUsed, wasmtesting.ErrMockContract
+				}
+			},
+			wasmtesting.ErrMockContract,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupWasmWithMockVM()
+
+			newCodeHash = storeWasmCode(suite, newByteCode)
+
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
+
+			oldClientState, ok := endpoint.GetClientState().(*types.ClientState)
+			suite.Require().True(ok)
+
+			expClientState = oldClientState
+
+			tc.malleate()
+
+			res, err := GetSimApp(suite.chainA).WasmClientKeeper.MigrateContract(suite.chainA.GetContext(), msg)
+
+			// updated client state
+			clientState, ok := endpoint.GetClientState().(*types.ClientState)
+			suite.Require().True(ok)
+
+			if tc.expError == nil {
+				expClientState.CodeHash = newCodeHash
+
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+
+				suite.Require().Equal(expClientState, clientState)
+			} else {
+				suite.Require().ErrorIs(err, tc.expError)
+				suite.Require().Nil(res)
+
+				suite.Require().Equal(expClientState, clientState)
```

Comment:
state checks are unnecessary for error conditions as you can assume baseapp will revert any potential state changes

Reply:
removed